drivers/pwm/pwm_sam0.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2020 Google LLC.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * PWM driver using the SAM0 Timer/Counter (TCC) in Normal PWM (NPWM) mode.
+ * Supports the SAMD21 and SAMD5x series.
+ */
+
+#define DT_DRV_COMPAT atmel_sam0_tcc_pwm
+
+#include <device.h>
+#include <errno.h>
+#include <drivers/pwm.h>
+#include <soc.h>
+
+/* Static configuration */
+struct pwm_sam0_config {
+	Tcc *regs;
+	uint8_t channels;
+	uint8_t counter_size;
+	uint16_t prescaler;
+	uint32_t freq;
+
+#ifdef MCLK
+	volatile uint32_t *mclk;
+	uint32_t mclk_mask;
+	uint16_t gclk_id;
+#else
+	uint32_t pm_apbcmask;
```

Comment:
In case of SAM D/E 5x, the TCC peripherals are connected to APB B, C and D. This should be reworked to reflect/support that.

Reply:
Hi @stephanosio could you tell me more?  Are you talking about the gclk_id block (used on the SAMD51) or the pm_apbcmask block (used on the SAMD21)?

For the SAMD51 it doesn't matter what bus the peripheral is on as the clocks are bus independent.

Reply:
> For the SAMD51 it doesn't matter what bus the peripheral is on as the clocks are bus independent.

ah, yes. you are right. I was looking at `MCLK_APBCMASK`.